api/v1beta2/awscluster_webhook.go

```diff
@@ -246,13 +246,20 @@ func (r *AWSCluster) validateNetwork() field.ErrorList {
 	return allErrs
 }
 
-func (r *AWSCluster) validateControlPlaneLBIngressRules() field.ErrorList {
+func (r *AWSCluster) validateControlPlaneLB() field.ErrorList {
 	var allErrs field.ErrorList
 
 	if r.Spec.ControlPlaneLoadBalancer == nil {
 		return allErrs
 	}
 
+	// Additional listeners are only supported for NLBs.
+	if len(r.Spec.ControlPlaneLoadBalancer.AdditionalListeners) > 0 {
+		if r.Spec.ControlPlaneLoadBalancer.LoadBalancerType != LoadBalancerTypeNLB {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "controlPlaneLoadBalancer", "additionalNetworkListeners"), r.Spec.ControlPlaneLoadBalancer.AdditionalListeners, "additional listeners are only supported for NLB load balancers"))
```

Comment:
This path does not align to the json tag
```suggestion
			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "controlPlaneLoadBalancer", "additionalListeners"), r.Spec.ControlPlaneLoadBalancer.AdditionalListeners, "additional listeners are only supported for NLB load balancers"))
```